[BUG]Build error with GCC9 in smart_amp_test.c

**Describe the bug**
A new build bug when using GCC9
```
/home/sof/work/sof.git/src/audio/smart_amp_test.c: In function 'smart_amp_set_config':
/home/sof/work/sof.git/src/audio/smart_amp_test.c:93:39: error: taking address of packed member of 'struct sof_abi_hdr' may result in an unaligned pointer value [-Werror=address-of-packed-member]
   93 |  cfg = (struct sof_smart_amp_config *)cdata->data->data;
```


**To Reproduce**
```
docker pull thesofproject/sof:20200728
docker tag thesofproject/sof:20200728 sof
./scripts/docker-run.sh ./scripts/xtensa-build-all.sh cnl
```

**Reproduction Rate**
10/10

**Expected behavior**
The GCC build should pass

--- a/src/audio/smart_amp_test.c
+++ b/src/audio/smart_amp_test.c
@@ -90,7 +90,8 @@ static int smart_amp_set_config(struct comp_dev *dev,
 	size_t bs;
 
 	/* Copy new config, find size from header */
-	cfg = (struct sof_smart_amp_config *)cdata->data->data;
+	cfg = (struct sof_smart_amp_config *)
+	       ASSUME_ALIGNED(cdata->data->data, sizeof(uint32_t));
 	bs = cfg->size;
 
 	comp_dbg(dev, "smart_amp_set_config(), actual blob size = %u, expected blob size = %u",

